lib/chef/json_compat.rb

```diff
@@ -138,6 +139,8 @@ def class_for_json_class(json_class)
           Chef::Node
         when CHEF_ROLE
           Chef::Role
+        when CHEF_RUNLISTEXPANSION
+          Chef::RunListExpansion
```

Comment:
You should only need to add this if you are setting `"json_class" => "Chef::RunListExpansion"` in your JSON representation. If you did that, then `Chef::JSONCompat.from_json(string_o_json)` would return a `Chef::RunListExpansion` object instead of a Hash. See for example, nodes: https://github.com/chef/chef/blob/8f65f75b2ae48ed33cfce1853edcdcc1a949b9c3/lib/chef/node.rb#L504

That said, I would recommend against adding the `json_class` thing because while it's sometimes convenient, sometimes it's not and it's difficult to turn it off when you want to. If maintaining compatibility weren't so hard, we'd have ripped out all of this long ago.